src/entries/popup/hooks/useInfiniteTransactionList.ts

```diff
@@ -218,8 +219,117 @@ export const useInfiniteTransactionList = ({
 
   const refetchTransactions = async () => {
     setManuallyRefetching(true);
-    await refetch();
-    setManuallyRefetching(false);
+    try {
```

Comment:
Am spooked by the complexity / potential fragility here. What does this do that is not handled by existing failure/success states for page fetches? I assume some of the logic should actually be upstream in the query fetcher, but would need to understand more about what's happening here.